src/main/scala/io/scalac/wtf/domain/UserService.scala

```diff
@@ -0,0 +1,37 @@
+package io.scalac.wtf.domain
+
+import io.scalac.wtf.domain.User.{UserAlreadyExists, ValidationError}
+import slick.dbio.DBIO
+import slick.driver.H2Driver.api._
+
+import cats.data.Validated.{invalidNel, valid}
+import cats.data.{NonEmptyList, ValidatedNel}
+import cats.implicits._
+import slick.lifted.MappedTo
+
+import scala.concurrent.ExecutionContext
+
+object UserService {
+
+  import scala.concurrent.ExecutionContext.Implicits.global
+
+  /*
+  //TODO: Move this somewhere else
+  //Evidence for DBIO being a cats Monad
+  implicit val dbioMonad = new Functor[DBIO] with Applicative[DBIO] with Monad[DBIO] {
+    def pure[A](a: A): DBIO[A] = DBIO.successful(a)
+    def flatMap[A, B](fa: DBIO[A])(f: A => DBIO[B]) = fa.flatMap(f)
+  }
+*/
+  def createUser(createdUser: User)(implicit executionContext: ExecutionContext): DBIO[ValidatedNel[ValidationError, UserId]] = {
+    val validation = User.validateUser(createdUser.email, createdUser.password)
+
+    UserRepository.findByEmail(createdUser.email).flatMap {
```

Comment:
This whole `createUser` function looks too complicated. If we now try to add some more stuff to it - like sending a confirmation email - this is going to be a mess. I'll try to propose a solution myself in the evening. Unless you have some ideas. The problems I see:

- too much code - for example the line `val userExistsValidated: ValidatedNel[ValidationError, UserId] = invalidNel(UserAlreadyExists)` does virtually nothing and yet it takes a lot of space - maybe there's some other way to writing it
- no clear step by step path of what's going on